### ISSUE TITLE
Run Supported Towers capabilities

### ISSUE THREAD

**Issue by wwieder** [ESCOMP/CTSM#1487]
Expand capabilities from `run_neon` to simulate additional supported tower sites (e.g. PLUMBER2 & AmeriFlux for RUBISCO).

Since we already have surface datasets and datm input for additional tower sites we should be able to build on the `run_neon` script to automate the generation of spinup, postAD and transient simulations for different tower sites.

From our conversation at the SE meeting it sounds like this could be done by creating the appropriate user_mods directories for each network / tower site and then adding some additional functionality to a more generic `run_tower` script that facilitates running one or multiple supported tower sites.

@olyson noted that PUMBER has some additional complications because all tower sites:
- Run with local time, meaning they have to initialize the model at different starting times
- Span different time periods, length of records
- Forcing data are at 30 or 60 minute time intervals
- Use LAI stream files for SP simulations (these have also been created)
@ekluzek suggested this can all be handled by the user_mod directories.

This issue is closely related to #1354, but has more/  updated specifics

**Comment by ekluzek** [comment]
Adding next to this so that we decide who will be working on this and about when we want this to happen.

**Comment by wwieder** [comment]
I don't see us doing this in the short term, so I'll close this issue for now.

**Comment by wwieder** [comment]
I'm reopening this issue because it seems that we'd like to broaden capabilities of single point runs to include PLUMBER2 sites.  This can now build on the subset_data and run_neon capabilities we've already created and can hopefully be extended to this broader set of towers that are being processed through ONEFLUX and evaluated with modelevaluation.org.

**Comment by wwieder** [comment]
Assigning Keith and Sam to this for now, as they were in our meeting yesterday, but we should discuss prioritization for this and who's best suited to do the work.

**Comment by wwieder** [comment]
@ekluzek noted that we want to bring in testing for this python script.  The high level design would to use the same infrastructure to call either NEON or PLUMBER sites.  

We can also use this as an opportunity to clean up some of the logic / duplication of features handled by this python script vs. those in usermod_dirs. I'll set up a meeting to work on this code design.

A timeline for this could be to complete this by December 2023.

**Comment by ekluzek** [comment]
This is part of the work the team will be doing in our hackathon to get this going for tower sites.

Myself and the following...

@adrifoster 
@olyson 
@slevis-lmwg
@TeaganKing 
@wwieder

**Comment by slevis-lmwg** [comment]
Corresponding notes in this [google doc.](https://docs.google.com/document/d/1i57JZgu6vWtdvr8rNMEnrtNIMIl8hB4jUeE4UvxNh38/edit#heading=h.75chhclse36y)

**Comment by ekluzek** [comment]
One thing we decided today is that we'll add an additional recognized option to CLM_USRDAT_NAME for PLUMBER2. The NEON options also have a NEONSITE XML variable for it's list of sites, similarly there will be a PLUMBER2SITE XML variable for the PLUMBER2 list of sites.

**Comment by wwieder** [comment]
This seems to be subsumed in these two tower site PRs

#2186
#2155

**Comment by TeaganKing** [comment]
As [PR #2135](https://github.com/ESCOMP/CTSM/pull/2315) is about to be merged in, I wanted to make sure we document the expected changes in a future PR to run supported towers capabilities:
- Build abstract TowerSite class
- Generic class
- Add NEON class with NEON-specific behavior
- Create new methods and unit tests
- Add PLUMBER class and behaviors
- Integration testing

**Comment by wwieder** [comment]
Is there more we should expect here beyond #2406 on a CLM6 timeline, @ekluzek, @adrifoster, @TeaganKing?  It seems like many of the user interface ideas are subsumed in new issues and maybe we can close this issue once b4b-dev gets merged?

**Comment by ekluzek** [comment]
Yeah, I think this is covered in #2406 and previous PR's. It looks like it isn't all setup in user-mods, but I think run_tower handles some of the details talked about at the top. I looked through the google doc, Sam L. mentioned above and most of it seems to be handled as well.

The one thing worthwhile seemed to be to do some work in build-namelist to remove some of the hacky shell_commands logic for some of this. But, I'll make that into it's own issue.

**Comment by ekluzek** [comment]
OK #2951 is the issue that has the one thing that would be useful to do. I'll set this up to autoclose when b4b-dev comes to master.

**Comment by TeaganKing** [comment]
Thanks everyone for your work on this. I agree that we can close this once b4b-dev is merged into master. The other anticipated changes above have already been made in that or other PRs.
